WooCommerce/WooCommerceTests/POS/ViewModels/CartViewModelTests.swift

```diff
@@ -0,0 +1,139 @@
+import XCTest
+import Combine
+@testable import WooCommerce
+@testable import protocol Yosemite.POSItem
+@testable import struct Yosemite.POSProduct
+
+final class CartViewModelTests: XCTestCase {
+
+    private var orderStageSubject: PassthroughSubject<PointOfSaleDashboardViewModel.OrderStage, Never>!
+    private var sut: CartViewModel!
+
+    override func setUp() {
+        super.setUp()
+        orderStageSubject = PassthroughSubject<PointOfSaleDashboardViewModel.OrderStage, Never>()
+        sut = CartViewModel(orderStage: orderStageSubject.eraseToAnyPublisher())
+    }
+
+    override func tearDown() {
+        orderStageSubject = nil
+        sut = nil
+        super.tearDown()
+    }
+
+    func test_canDeleteItemsFromCart_when_orderStage_is_building_then_returns_true() {
+        // Given/When
+        XCTAssertEqual(sut.canDeleteItemsFromCart, true, "Initial state")
+        orderStageSubject.send(.finalizing)
+
+        // Then
+        XCTAssertEqual(sut.canDeleteItemsFromCart, false)
+    }
+
+    func test_canDeleteItemsFromCart_when_orderStage_is_finalizing_then_returns_false() {
+        // Given/When
+        orderStageSubject.send(.finalizing)
+
+        // Then
+        XCTAssertEqual(sut.canDeleteItemsFromCart, false)
+    }
+
+    func test_cart_when_submitCart_is_invoked_then_cartSubmissionPublisher_emits_cart_items() {
+        // Given
+        var cancellables: Set<AnyCancellable> = []
+        let item = Self.makeItem()
+        let anotherItem = Self.makeItem()
+
+        // When
+        sut.addItemToCart(item)
+        sut.addItemToCart(anotherItem)
+        sut.cartSubmissionPublisher.sink(receiveValue: { cartItems in
+            // Then
+            XCTAssertEqual(cartItems.count, 2)
+        })
+        .store(in: &cancellables)
+
+        sut.submitCart()
+    }
+
+    func test_cart_when_addItemToCart_is_invoked_then_adds_item_to_cart() {
+        // Given
+        XCTAssertTrue(sut.itemsInCart.isEmpty, "Initial state")
+        let item = Self.makeItem()
+
+        // When
+        sut.addItemToCart(item)
+
+        // Then
+        XCTAssertTrue(sut.itemsInCart.isNotEmpty)
+    }
+
+    func test_removeItemFromCart() {
+        /* TODO:
+         https://github.com/woocommerce/woocommerce-ios/issues/13209
+         The unique UUID for CartItem is set on init, but CartItem is only internal to addItemToCart()
+         We need to extract this to a separate function and assure that ID's are correct,
+         otherwise the UUID's for testing won't match
+         */
```

Comment:
For this test case, I wonder if we could first add some items by using the cart item from `itemsInCart`. I added a simple test case in https://github.com/woocommerce/woocommerce-ios/pull/13191/commits/e15efe2ec908f30a63675e8ff334624a6819d8b5, feel free to add other edge cases in the TODO issue.

Reply:
Yes, initially I wanted to test only by `id` to assure that we remove the exact "line item" if a cart has multiple of the same `productID`, so if we happen to change the implementation, we wouldn't get a false positive from tests if we happen to have just another item with the same name or property in the list. 

That said I think testing it like this and adding the followup is good enough for the moment (better than having no test case!). Thanks for tackling it 🙇